integration-tests: fail when running with jdk11

1. ASSERT: make sure that you're using jdk11
2. `mvn clean verify -Pintegration-tests`

Result:
```
Failed tests:   testStartServerTwice(org.jboss.tools.rsp.itests.wildfly.WildFlyLaunchingTest): Waiting for server state to change to started timed out
  testStartServer(org.jboss.tools.rsp.itests.wildfly.WildFlyLaunchingTest): Waiting for server state to change to started timed out
  testStartServer(org.jboss.tools.rsp.itests.wildfly.WildFlyPublishingTest): Waiting for server state to change to a non-null value
```

This issue still exists:

```
[INFO] Results:
[INFO]
[ERROR] Failures:
[ERROR] org.jboss.tools.rsp.itests.wildfly.WildFlyLaunchingTest.testStartServer(org.jboss.tools.rsp.itests.wildfly.WildFlyLaunchingTest)
[ERROR]   Run 1: WildFlyLaunchingTest.testStartServer:241->waitForServerState:321 Waiting for server state to change to started timed out
[ERROR]   Run 2: WildFlyLaunchingTest.testStartServer:241->waitForServerState:321 Waiting for server state to change to started timed out
[INFO]
[ERROR] org.jboss.tools.rsp.itests.wildfly.WildFlyLaunchingTest.testStartServerTwice(org.jboss.tools.rsp.itests.wildfly.WildFlyLaunchingTest)
[ERROR]   Run 1: WildFlyLaunchingTest.testStartServerTwice:257->waitForServerState:321 Waiting for server state to change to started timed out
[ERROR]   Run 2: WildFlyLaunchingTest.testStartServerTwice:257->waitForServerState:321 Waiting for server state to change to started timed out
[INFO]
[ERROR] org.jboss.tools.rsp.itests.wildfly.WildFlyLaunchingTest.testStopServer(org.jboss.tools.rsp.itests.wildfly.WildFlyLaunchingTest)
[ERROR]   Run 1: WildFlyLaunchingTest.testStopServer:275->waitForServerState:321 Waiting for server state to change to started timed out
[ERROR]   Run 2: WildFlyLaunchingTest.testStopServer:275->waitForServerState:321 Waiting for server state to change to started timed out
[INFO]
[ERROR] org.jboss.tools.rsp.itests.wildfly.WildFlyPublishingTest.testStartServer(org.jboss.tools.rsp.itests.wildfly.WildFlyPublishingTest)
[ERROR]   Run 1: WildFlyPublishingTest.testStartServer:60->waitForServerState:154 Waiting for server state to change to started timed out
[ERROR]   Run 2: WildFlyPublishingTest.testStartServer:60->waitForServerState:154 Waiting for server state to change to started timed out
[INFO]
[INFO]
[ERROR] Tests run: 110, Failures: 4, Errors: 0, Skipped: 0
```

--- a/Jenkinsfile
+++ b/Jenkinsfile
@@ -35,7 +35,7 @@ pipeline {
                 		
                         stage('Integration tests') {
                 			steps {
-                				sh 'mvn verify -B -Pintegration-tests -DskipTests=true -Dmaven.test.failure.ignore=true'
+                				sh 'mvn verify -B -Pintegration-tests -Dwildfly-14 -DskipTests=true -Dmaven.test.failure.ignore=true'
                 			}
                         }
                 		
@@ -76,7 +76,7 @@ pipeline {
                 		
                         stage('Integration tests') {
                 			steps {
-                				sh 'mvn verify -B -Pintegration-tests -Dwildfly-15 -DskipTests=true -Dmaven.test.failure.ignore=true'
+                				sh 'mvn verify -B -Pintegration-tests -DskipTests=true -Dmaven.test.failure.ignore=true'
                         	}
                         }
     	            }

--- a/distribution/integration-tests/pom.xml
+++ b/distribution/integration-tests/pom.xml
@@ -82,7 +82,7 @@
 			<id>wildfly-14</id>
 			<activation>
 				<property>
-					<name>!wildfly-15</name>
+					<name>wildfly-14</name>
 				</property>
 			</activation>
 			<build>
@@ -126,7 +126,7 @@
 			<id>wildfly-15</id>
 			<activation>
 				<property>
-					<name>wildfly-15</name>
+					<name>!wildfly-14</name>
 				</property>
 			</activation>
 			<build>

--- a/distribution/integration-tests/pom.xml
+++ b/distribution/integration-tests/pom.xml
@@ -254,5 +254,5 @@
 			<version>4.12</version>
 			<scope>test</scope>
 		</dependency>
-	</dependencies>
+  	</dependencies>
 </project>

--- a/distribution/integration-tests/src/test/java/org/jboss/tools/rsp/itests/RSPCase.java
+++ b/distribution/integration-tests/src/test/java/org/jboss/tools/rsp/itests/RSPCase.java
@@ -12,12 +12,15 @@ import static org.junit.Assert.fail;
 
 import java.nio.file.Paths;
 import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
+import java.util.concurrent.CompletableFuture;
 
 import org.jboss.tools.rsp.api.RSPServer;
 import org.jboss.tools.rsp.api.dao.DiscoveryPath;
 import org.jboss.tools.rsp.api.dao.ServerAttributes;
 import org.jboss.tools.rsp.api.dao.ServerBean;
+import org.jboss.tools.rsp.api.dao.ServerHandle;
 import org.jboss.tools.rsp.api.dao.ServerType;
 import org.jboss.tools.rsp.api.dao.Status;
 import org.jboss.tools.rsp.itests.util.DummyClientLauncher;
@@ -85,4 +88,16 @@ public abstract class RSPCase {
 		ServerAttributes serverAttr = new ServerAttributes(bean.getServerAdapterTypeId(), id, attr);
 		return serverProxy.createServer(serverAttr).get().getStatus();
 	}
+
+	protected void deleteServer(String id) throws Exception {
+		CompletableFuture<List<ServerHandle>> handles = serverProxy.getServerHandles();
+		ServerHandle handle = handles.get().stream()
+			.filter(server -> id.equals(server.getId()))
+			.findFirst().orElse(null);
+		if (handle == null) {
+			return;
+		}
+		serverProxy.deleteServer(handle);
+	}
+
 }

--- a/distribution/integration-tests/src/test/java/org/jboss/tools/rsp/itests/util/ServerStateUtil.java
+++ b/distribution/integration-tests/src/test/java/org/jboss/tools/rsp/itests/util/ServerStateUtil.java
@@ -1,0 +1,64 @@
+/*******************************************************************************
+ * Copyright (c) 2019 Red Hat, Inc. Distributed under license by Red Hat, Inc.
+ * All rights reserved. This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is
+ * available at http://www.eclipse.org/legal/epl-v20.html
+ * 
+ * Contributors: Red Hat, Inc.
+ ******************************************************************************/
+package org.jboss.tools.rsp.itests.util;
+
+import org.jboss.tools.rsp.api.ServerManagementAPIConstants;
+import org.jboss.tools.rsp.api.dao.ServerState;
+
+public class ServerStateUtil {
+
+	private static final long WAIT_FOR_SERVERSTATE = 3000;
+    private static final int NO_STATE = -1;
+
+	public static String toStateString(ServerState state) {
+		if (state == null) {
+			return "null state";
+		}
+		
+		return toString(state.getState());
+	}
+
+	public static String toString(int state) {
+		switch(state) {
+		case ServerManagementAPIConstants.STATE_STARTING:
+			return "starting";
+		case ServerManagementAPIConstants.STATE_STARTED:
+			return "started";
+		case ServerManagementAPIConstants.STATE_STOPPING:
+			return "stopping";
+		default:
+		case ServerManagementAPIConstants.STATE_UNKNOWN:
+			return "unknown";
+			
+		}
+	}
+
+    public static ServerState waitForNonNullServerState(int attempts, DummyClient client) throws Exception {
+    	return waitForServerState(NO_STATE, attempts, client);
+    }
+
+    public static ServerState waitForServerState(int expected, int attempts, DummyClient client) throws Exception {
+        int tries = attempts;
+        
+        while(tries > 0) {
+            tries--;
+            ServerState s = client.getStateObject();
+			if (s != null) {
+				if (expected == NO_STATE
+						|| expected == s.getState()) {
+					client.clearState();
+					return s;
+				}
+			}
+            System.out.println("Server state: " + ServerStateUtil.toStateString(s));
+            Thread.sleep(WAIT_FOR_SERVERSTATE);
+        }
+        throw new AssertionError("Waiting for server state to change to " + expected + " timed out");
+    }
+}

--- a/distribution/integration-tests/src/test/java/org/jboss/tools/rsp/itests/wildfly/WildFlyLaunchingTest.java
+++ b/distribution/integration-tests/src/test/java/org/jboss/tools/rsp/itests/wildfly/WildFlyLaunchingTest.java
@@ -8,9 +8,18 @@
  ******************************************************************************/
 package org.jboss.tools.rsp.itests.wildfly;
 
+import static org.jboss.tools.rsp.itests.util.ServerStateUtil.waitForServerState;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+
+import org.jboss.tools.rsp.api.ServerManagementAPIConstants;
 import org.jboss.tools.rsp.api.dao.Attributes;
 import org.jboss.tools.rsp.api.dao.CommandLineDetails;
 import org.jboss.tools.rsp.api.dao.LaunchAttributesRequest;
@@ -26,7 +35,6 @@ import org.jboss.tools.rsp.itests.RSPCase;
 import org.jboss.tools.rsp.itests.util.DummyClient;
 import org.junit.Test;
 
-import static org.junit.Assert.*;
 
 /**
  *
@@ -37,9 +45,6 @@ public class WildFlyLaunchingTest extends RSPCase {
     private static final String MODE_DEBUG = "debug";
 	private static final String MODE_RUN = "run";
 	private static final String STATUS_MESSAGE_OK = "ok";
-	private static final String STATE_STOPPED = "stopped";
-	private static final String STATE_STARTED = "started";
-	private static final long WAIT_FOR_SERVERSTATE = 3000;
 
 	private final DummyClient client = launcher.getClient();
 
@@ -238,10 +243,10 @@ public class WildFlyLaunchingTest extends RSPCase {
         
         assertEquals(0, response.getStatus().getSeverity());
         assertEquals(STATUS_MESSAGE_OK, response.getStatus().getMessage());
-        waitForServerState(STATE_STARTED, 10);
+        waitForServerState(ServerManagementAPIConstants.STATE_STARTED, 10, client);
         
         serverProxy.stopServerAsync(new StopServerAttributes("wildfly3", true)).get();
-        waitForServerState(STATE_STOPPED, 10);
+        waitForServerState(ServerManagementAPIConstants.STATE_STOPPED, 10, client);
     }
 
     @Test
@@ -254,13 +259,13 @@ public class WildFlyLaunchingTest extends RSPCase {
                 new ServerAttributes(wildflyType.getId(), "wildfly4", attr), MODE_RUN);
         
         serverProxy.startServerAsync(params).get();
-        waitForServerState(STATE_STARTED, 10);
+        waitForServerState(ServerManagementAPIConstants.STATE_STARTED, 10, client);
         StartServerResponse response = serverProxy.startServerAsync(params).get();
         
         assertEquals(Status.CANCEL, response.getStatus().getSeverity());
         
         serverProxy.stopServerAsync(new StopServerAttributes("wildfly4", true)).get();
-        waitForServerState(STATE_STOPPED, 10);
+        waitForServerState(ServerManagementAPIConstants.STATE_STOPPED, 10, client);
     }
 
     @Test
@@ -272,10 +277,10 @@ public class WildFlyLaunchingTest extends RSPCase {
                 new ServerAttributes(wildflyType.getId(), "wildfly5", attr), MODE_RUN);
         
         serverProxy.startServerAsync(params).get();
-        waitForServerState(STATE_STARTED, 10);
+        waitForServerState(ServerManagementAPIConstants.STATE_STARTED, 10, client);
         
         Status status = serverProxy.stopServerAsync(new StopServerAttributes("wildfly5", false)).get();
-        waitForServerState(STATE_STOPPED, 10);
+        waitForServerState(ServerManagementAPIConstants.STATE_STOPPED, 10, client);
         
         assertEquals(Status.OK, status.getSeverity());
         assertEquals(STATUS_MESSAGE_OK, status.getMessage());
@@ -308,16 +313,4 @@ public class WildFlyLaunchingTest extends RSPCase {
         assertEquals(Status.ERROR, status.getSeverity());
     }
     
-    private void waitForServerState(String serverState, int tries) throws Exception {
-        int remaining = tries;
-
-		while (remaining > 0) {
-			remaining--;
-            if (serverState.equals(client.getStateString())) {
-                return;
-            }
-            Thread.sleep(WAIT_FOR_SERVERSTATE);
-        }
-        throw new AssertionError("Waiting for server state to change to " + serverState + " timed out");
-    }
 }

--- a/distribution/integration-tests/src/test/java/org/jboss/tools/rsp/itests/wildfly/WildFlyPublishingTest.java
+++ b/distribution/integration-tests/src/test/java/org/jboss/tools/rsp/itests/wildfly/WildFlyPublishingTest.java
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2018 Red Hat, Inc. Distributed under license by Red Hat, Inc.
+ * Copyright (c) 2018-2019 Red Hat, Inc. Distributed under license by Red Hat, Inc.
  * All rights reserved. This program is made available under the terms of the
  * Eclipse Public License v2.0 which accompanies this distribution, and is
  * available at http://www.eclipse.org/legal/epl-v20.html
@@ -8,8 +8,12 @@
  ******************************************************************************/
 package org.jboss.tools.rsp.itests.wildfly;
 
+import static org.jboss.tools.rsp.itests.util.ServerStateUtil.waitForNonNullServerState;
+import static org.jboss.tools.rsp.itests.util.ServerStateUtil.waitForServerState;
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
 import static org.junit.Assert.fail;
 
 import java.io.File;
@@ -19,6 +23,7 @@ import java.nio.file.Path;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.concurrent.ExecutionException;
 
 import org.jboss.tools.rsp.api.ServerManagementAPIConstants;
 import org.jboss.tools.rsp.api.dao.DeployableReference;
@@ -34,6 +39,8 @@ import org.jboss.tools.rsp.api.dao.StopServerAttributes;
 import org.jboss.tools.rsp.itests.RSPCase;
 import org.jboss.tools.rsp.itests.util.DeploymentGeneration;
 import org.jboss.tools.rsp.itests.util.DummyClient;
+import org.junit.After;
+import org.junit.Before;
 import org.junit.Test;
 
 /**
@@ -41,36 +48,44 @@ import org.junit.Test;
  * @author jrichter
  */
 public class WildFlyPublishingTest extends RSPCase {
-    
-    private static final String WAR_FILENAME = "test9.war";
+
+	private static final String SERVER_ID = "wildfly6a";
+	private static final String WAR_FILENAME = "test9.war";
     private final DummyClient client = launcher.getClient();
-    
+
+    @Before
+    public void before() throws Exception {
+        createServer(WILDFLY_ROOT, SERVER_ID);
+    }
+
+    @After
+    public void after() throws Exception {
+    	deleteServer(SERVER_ID);
+    }
+
     @Test
-    public void testStartServer() throws Exception {
-        createServer(WILDFLY_ROOT, "wildfly6a");
+    public void testStartServer() throws Exception, ExecutionException {
         Map<String, Object> attr = new HashMap<>();
         attr.put("server.home.dir", WILDFLY_ROOT);
         LaunchParameters params = new LaunchParameters(
-                new ServerAttributes(wildflyType.getId(), "wildfly6a", attr), "run");
+                new ServerAttributes(wildflyType.getId(), SERVER_ID, attr), "run");
         
         StartServerResponse response = serverProxy.startServerAsync(params).get();
         
-        assertEquals(0, response.getStatus().getSeverity());
+        assertTrue("server wasnt successfully started: " + response.getStatus().getMessage(), response.getStatus().isOK());
         assertEquals("ok", response.getStatus().getMessage());
-        waitForServerState("started", 10);
+        waitForServerState(ServerManagementAPIConstants.STATE_STARTED, 20, client);
         
-        ServerHandle handle = new ServerHandle("wildfly6a", wildflyType);
+        ServerHandle handle = new ServerHandle(SERVER_ID, wildflyType);
         File war = createWar();
         DeployableReference deployable = new DeployableReference(war.getAbsolutePath(), war.getAbsolutePath());
         ServerDeployableReference req = new ServerDeployableReference(handle, deployable);
         serverProxy.addDeployable(req);
         
-        ServerState state = waitForNonNullServerState(5);
+        ServerState state = waitForNonNullServerState(20, client);
         assertNotNull(state);
-        List<DeployableState> list = state.getDeployableStates();
-        assertNotNull(list);
-        assertEquals(1, list.size());
-        DeployableState ds1 = list.get(0);
+        DeployableState ds1 = getDeployableState(deployable, state.getDeployableStates());
+        assertNotNull("state for deployable " + deployable.getPath() + " wasnt found.", ds1);
         assertNotNull(ds1);
         assertEquals(ServerManagementAPIConstants.PUBLISH_STATE_ADD, ds1.getPublishState());
         assertEquals(ServerManagementAPIConstants.STATE_UNKNOWN, ds1.getState());
@@ -78,36 +93,29 @@ public class WildFlyPublishingTest extends RSPCase {
         // TODO publish
         PublishServerRequest pubReq = new PublishServerRequest(handle, ServerManagementAPIConstants.PUBLISH_FULL);
         serverProxy.publish(pubReq);
-        state = waitForNonNullServerState(5);
+        state = waitForNonNullServerState(20, client);
         assertNotNull(state);
-        list = state.getDeployableStates();
-        assertNotNull(list);
-        assertEquals(1, list.size());
-        ds1 = list.get(0);
+        ds1 = getDeployableState(deployable, state.getDeployableStates());
+        assertNotNull("state for deployable " + deployable.getPath() + " wasnt found.", ds1);
         assertNotNull(ds1);
         assertEquals(ServerManagementAPIConstants.PUBLISH_STATE_NONE, ds1.getPublishState());
         assertEquals(ServerManagementAPIConstants.STATE_STARTED, ds1.getState());
         
         // TODO touch the file
         war.setLastModified(System.currentTimeMillis());
-        state = waitForNonNullServerState(5);
+        state = waitForNonNullServerState(20, client);
         assertNotNull(state);
-        list = state.getDeployableStates();
-        assertNotNull(list);
-        assertEquals(1, list.size());
-        ds1 = list.get(0);
+        ds1 = getDeployableState(deployable, state.getDeployableStates());
+        assertNotNull("state for deployable " + deployable.getPath() + " wasnt found.", ds1);
         assertNotNull(ds1);
         assertEquals(ServerManagementAPIConstants.PUBLISH_STATE_INCREMENTAL, ds1.getPublishState());
         assertEquals(ServerManagementAPIConstants.STATE_STARTED, ds1.getState());
 
-        
         serverProxy.removeDeployable(new ServerDeployableReference(handle,  deployable));
-        state = waitForNonNullServerState(5);
+        state = waitForNonNullServerState(20, client);
         assertNotNull(state);
-        list = state.getDeployableStates();
-        assertNotNull(list);
-        assertEquals(1, list.size());
-        ds1 = list.get(0);
+        ds1 = getDeployableState(deployable, state.getDeployableStates());
+        assertNotNull("state for deployable " + deployable.getPath() + " wasnt found.", ds1);
         assertNotNull(ds1);
         assertEquals(ServerManagementAPIConstants.PUBLISH_STATE_REMOVE, ds1.getPublishState());
         assertEquals(ServerManagementAPIConstants.STATE_STARTED, ds1.getState());
@@ -115,14 +123,13 @@ public class WildFlyPublishingTest extends RSPCase {
         // TODO publish
         pubReq = new PublishServerRequest(handle, ServerManagementAPIConstants.PUBLISH_FULL);
         serverProxy.publish(pubReq);
-        state = waitForNonNullServerState(5);
+        state = waitForNonNullServerState(20, client);
         assertNotNull(state);
-        list = state.getDeployableStates();
-        assertNotNull(list);
-        assertEquals(0, list.size());
+        ds1 = getDeployableState(deployable, state.getDeployableStates());
+        assertNull("state for deployable " + deployable.getPath() + " was still found but shouldn't.", ds1);
                 
-        serverProxy.stopServerAsync(new StopServerAttributes("wildfly6a", true)).get();
-        waitForServerState("stopped", 10);
+        serverProxy.stopServerAsync(new StopServerAttributes(SERVER_ID, true)).get();       
+        waitForServerState(ServerManagementAPIConstants.STATE_STOPPED, 10, client);
     }
 
 	protected File createWar() {
@@ -137,35 +144,15 @@ public class WildFlyPublishingTest extends RSPCase {
 		} catch (IOException e) {}
 		return war != null && war.exists() && war.isFile() ? war : null;
 	}
-	
-    private void waitForServerState(String serverState, int timeout) throws Exception {
-        int tries = timeout;
-        
-        while(tries > 0) {
-            tries--;
-            String s = client.getStateString();
-            System.out.println("State String: " + s);
-            if (serverState.equals(s)) {
-            	client.clearState();
-                return;
-            }
-            Thread.sleep(1000);
-        }
-        throw new AssertionError("Waiting for server state to change to " + serverState + " timed out");
-    }
 
-    private ServerState waitForNonNullServerState(int timeout) throws Exception {
-        int tries = timeout;
-        
-        while(tries > 0) {
-            tries--;
-        	ServerState ret = client.getStateObject();
-            if (ret != null) {
-            	client.clearState();
-                return ret;
-            }
-            Thread.sleep(1000);
-        }
-        throw new AssertionError("Waiting for server state to change to a non-null value");
-    }
+	public DeployableState getDeployableState(DeployableReference reference, List<DeployableState> states) {
+		if (states == null || states.isEmpty()) {
+			return null;
+		}
+		
+		return states.stream()
+				.filter(state -> reference.equals(state.getReference()))
+				.findFirst()
+				.orElse(null);
+	}    
 }

--- a/distribution/integration-tests/src/test/java/org/jboss/tools/rsp/itests/wildfly/WildFlyServerModelTest.java
+++ b/distribution/integration-tests/src/test/java/org/jboss/tools/rsp/itests/wildfly/WildFlyServerModelTest.java
@@ -8,19 +8,23 @@
  ******************************************************************************/
 package org.jboss.tools.rsp.itests.wildfly;
 
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+
 import org.jboss.tools.rsp.api.dao.Attribute;
 import org.jboss.tools.rsp.api.dao.Attributes;
 import org.jboss.tools.rsp.api.dao.ServerAttributes;
 import org.jboss.tools.rsp.api.dao.ServerHandle;
 import org.jboss.tools.rsp.api.dao.ServerType;
 import org.jboss.tools.rsp.api.dao.Status;
+import org.jboss.tools.rsp.eclipse.core.runtime.IStatus;
 import org.jboss.tools.rsp.itests.RSPCase;
 import org.junit.Test;
-
-import static org.junit.Assert.*;
 
 /**
  *
@@ -128,13 +132,14 @@ public class WildFlyServerModelTest extends RSPCase {
     
     @Test
     public void testCreateServer() throws Exception {
-        Status status = createServer(WILDFLY_ROOT, "wfly");
+    	String serverName = createUniqueName("wfly");
+    	Status status = createServer(WILDFLY_ROOT, serverName);
                 
-        assertEquals(0, status.getSeverity());
+        assertEquals(IStatus.OK, status.getSeverity());
         assertEquals("ok", status.getMessage());
         
         List<ServerHandle> handles = serverProxy.getServerHandles().get();
-        ServerHandle expected = new ServerHandle("wfly", wildflyType);
+        ServerHandle expected = new ServerHandle(serverName, wildflyType);
         assertTrue(handles.contains(expected));
     }
     
@@ -143,7 +148,7 @@ public class WildFlyServerModelTest extends RSPCase {
         ServerAttributes attr = new ServerAttributes("UNKNOWN", "fly", null);
         Status status = serverProxy.createServer(attr).get().getStatus();
         
-        assertEquals(Status.ERROR, status.getSeverity());
+        assertEquals(IStatus.ERROR, status.getSeverity());
         assertEquals("Server Type UNKNOWN not found", status.getMessage());
     }
     
@@ -151,19 +156,20 @@ public class WildFlyServerModelTest extends RSPCase {
     public void testCreateServerNull() throws Exception {
         Status status = serverProxy.createServer(null).get().getStatus();
         
-        assertEquals(Status.ERROR, status.getSeverity());
+        assertEquals(IStatus.ERROR, status.getSeverity());
         assertEquals(INVALID_PARAM, status.getMessage());
     }
     
     @Test
     public void testCreateServerTwice() throws Exception {
-        Status status1 = createServer(WILDFLY_ROOT, "wfly1");
-        Status status2 = createServer(WILDFLY_ROOT, "wfly1");
+    	String serverName = createUniqueName("wfly1");
+    	Status status1 = createServer(WILDFLY_ROOT, serverName);
+        Status status2 = createServer(WILDFLY_ROOT, serverName);
         
-        assertEquals(0, status1.getSeverity());
+        assertEquals(IStatus.OK, status1.getSeverity());
         assertEquals("ok", status1.getMessage());
-        assertEquals(Status.ERROR, status2.getSeverity());
-        assertEquals("Server with id wfly1 already exists.", status2.getMessage());
+        assertEquals(IStatus.ERROR, status2.getSeverity());
+        assertTrue(status2.getMessage().contains("already exists"));
     }
     
     @Test
@@ -173,7 +179,7 @@ public class WildFlyServerModelTest extends RSPCase {
         ServerHandle handle = new ServerHandle("wfly2", wildflyType);
         Status status = serverProxy.deleteServer(handle).get();
         
-        assertEquals(0, status.getSeverity());
+        assertEquals(IStatus.OK, status.getSeverity());
         assertEquals("ok", status.getMessage());
     }
     
@@ -182,7 +188,7 @@ public class WildFlyServerModelTest extends RSPCase {
         ServerHandle handle = new ServerHandle("wfly3", wildflyType);
         Status status = serverProxy.deleteServer(handle).get();
         
-        assertEquals(Status.ERROR, status.getSeverity());
+        assertEquals(IStatus.ERROR, status.getSeverity());
         assertEquals("Server not removed: wfly3", status.getMessage());
     }
     
@@ -191,7 +197,7 @@ public class WildFlyServerModelTest extends RSPCase {
         ServerHandle handle = new ServerHandle("wfly4", new ServerType("foo", "foo", "foo"));
         Status status = serverProxy.deleteServer(handle).get();
         
-        assertEquals(Status.ERROR, status.getSeverity());
+        assertEquals(IStatus.ERROR, status.getSeverity());
         assertEquals("Server not removed: wfly4", status.getMessage());
     }
     
@@ -199,7 +205,7 @@ public class WildFlyServerModelTest extends RSPCase {
     public void testDeleteServerNull() throws Exception {
         Status status = serverProxy.deleteServer(null).get();
         
-        assertEquals(Status.ERROR, status.getSeverity());
+        assertEquals(IStatus.ERROR, status.getSeverity());
         assertEquals(INVALID_PARAM, status.getMessage());
     }
     
@@ -212,5 +218,9 @@ public class WildFlyServerModelTest extends RSPCase {
         
         assertTrue(handles.contains(new ServerHandle("wfly5", wildflyType)));
         assertTrue(handles.contains(new ServerHandle("wfly6", wildflyType)));
+    }
+    
+    private static String createUniqueName(String prefix) {
+    	return prefix + System.currentTimeMillis();
     }
 }

